### PR TITLE
Store auth info

### PR DESCRIPTION
**Pull request by yujinjcho**
For https://github.com/yujinjcho/rcruitme/issues/17

### CODE REVIEW

File: app/models/daos/PasswordAuthInfoDAO.scala [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  def find(loginInfo: LoginInfo): Future[Option[PasswordInfo]] = {
+    val password: Option[String] = db.withConnection { implicit c =>
+      SQL("SELECT password FROM users WHERE email = {email}")
+        .on("email" -> loginInfo.providerKey)
+        .as(scalar[String].singleOpt)
+    }
+    Future.successful(password.map(p => PasswordInfo(new BCryptSha256PasswordHasher().id, p)))
+  }
+
+  def add(loginInfo: LoginInfo, authInfo: PasswordInfo): Future[PasswordInfo] = {
```
Comment (achiu8) [inline app/models/daos/PasswordAuthInfoDAO.scala]: What's the point of having this if it just calls `update`?

File: app/models/daos/PasswordAuthInfoDAO.scala [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            password = {password}
+          WHERE
+            email = {email}
+      """).on("password" -> authInfo.password, "email" -> loginInfo.providerKey).executeUpdate()
+    }
+    Future.successful(authInfo)
+  }
+
+  def save(loginInfo: LoginInfo, authInfo: PasswordInfo): Future[PasswordInfo] = {
+    find(loginInfo).flatMap {
```
Comment (achiu8) [inline app/models/daos/PasswordAuthInfoDAO.scala]: What's the point of having / calling `find` if we just discard the result and basically do the same thing in either case?

File: app/models/daos/SocialAuthInfoDAO.scala [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+) extends DelegableAuthInfoDAO[OAuth2Info] {
+
+  private val db = dbapi.database("default")
+
+  val TokenType = "Bearer"
+
+  def find(loginInfo: LoginInfo): Future[Option[OAuth2Info]] = {
+    val password: Option[(String,Int)] = db.withConnection { implicit c =>
+      SQL("SELECT google_token, token_expiry FROM users WHERE email = {email}")
+        .on("email" -> loginInfo.providerKey)
```
Comment (achiu8) [inline app/models/daos/SocialAuthInfoDAO.scala]: Isn't `loginInfo.providerKey` in this case `google_key`?

File: app/models/daos/PasswordAuthInfoDAO.scala (hunk 1 above)
Comment (yujinjcho) [reply app/models/daos/PasswordAuthInfoDAO.scala]: so the class extends `DelegableAuthInfoDAO` which extends [AuthInfoDAO](https://github.com/mohiva/play-silhouette/blob/master/silhouette-persistence/src/main/scala/com/mohiva/play/silhouette/persistence/daos/AuthInfoDAO.scala) which requires implementing these methods

File: app/models/daos/PasswordAuthInfoDAO.scala (hunk 2 above)
Comment (yujinjcho) [reply app/models/daos/PasswordAuthInfoDAO.scala]: actually could probably just call update since we're doing the same thing either way.

**Pull request merged**